ts/features/payments/onboarding/analytics/index.ts

```diff
@@ -1,15 +1,81 @@
 import { mixpanelTrack } from "../../../../mixpanel";
 import { buildEventProperties } from "../../../../utils/analytics";
+import { WalletOnboardingOutcomeEnum } from "../types/OnboardingOutcomeEnum";
 
 export type PaymentOnboardingAnalyticsProps = {
   payment_method_selected: string;
 };
 
-export const trackSuccessOnboardingPaymentMethod = (
+const trackSuccessOnboardingPaymentMethod = (
   props: Partial<PaymentOnboardingAnalyticsProps>
 ) => {
   void mixpanelTrack(
     "ADD_PAYMENT_METHOD_UX_SUCCESS",
     buildEventProperties("UX", "screen_view", props)
   );
 };
+
+const trackOnboardingPaymentMethodDenied = (
+  props: Partial<PaymentOnboardingAnalyticsProps>
+) => {
+  void mixpanelTrack(
+    "PAYMENT_ADD_METHOD_AUTHORIZATION_DENIED",
+    buildEventProperties("UX", "screen_view", props)
+  );
+};
+
+const trackAddOnboardingPaymentMethodCanceled = (
+  props: Partial<PaymentOnboardingAnalyticsProps>
+) => {
+  void mixpanelTrack(
+    "PAYMENT_ADD_METHOD_CANCELED_BY_USER",
+    buildEventProperties("UX", "screen_view", props)
+  );
+};
+
+const trackAddOnboardingPaymentMethodDuplicated = (
+  props: Partial<PaymentOnboardingAnalyticsProps>
+) => {
+  void mixpanelTrack(
+    "PAYMENT_ADD_METHOD_DUPLICATE_ERROR",
+    buildEventProperties("UX", "screen_view", props)
+  );
+};
+
+// This function will be used in the future when we will have 3DS
+// const trackOnboardingPaymentMethod3dsError = (
+//   props: Partial<PaymentOnboardingAnalyticsProps>
+// ) => {
+//   void mixpanelTrack(
+//     "PAYMENT_ADD_METHOD_3DS_ERROR",
+//     buildEventProperties("UX", "screen_view", props)
+//   );
+// };
+
+export const trackFailureOnboardingPaymentMethod = (
```

Comment:
Since in this function is handled also the success case, maybe it's better to rename this function in something like `trackAddOnboardingPaymentMethod`, what do you think?